zstacklib/zstacklib/utils/file_metadata_handler.py

```
@@ -0,0 +1,209 @@
+            return {}
+
+    def _do_get(self, metadataPath):
+        with self._get_path_lock(metadataPath):
+            if not os.path.isfile(metadataPath):
+                tmp_path = metadataPath + ".tmp"
+                if os.path.isfile(tmp_path):
+                    try:
+                        os.rename(tmp_path, metadataPath)
+                        _fsync_directory(metadataPath)
+                        logger.info("recovered tmp file to %s during get", metadataPath)
+                    except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)